enhancements/cluster-logging/fine-grained-per-namespace-logs-access.md

```diff
@@ -0,0 +1,249 @@
+---
+title: fine-grained-logs-access
+
+authors:
+  - "@aminesnow"
+
+reviewers:
+  - "@periklis"
+  - "@shwetaap"
+  - "@alanconway"
+  - "@xperimental"
+  - "@jcantrill"
+
+approvers:
+  - "@periklis"
+
+creation-date: 2023-05-31
+last-updated: 2023-05-31
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/LOG-4020
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Openshift Logging: Fine grained access to logs
+
+## Summary
+
+This enhancement proposal seeks to outline the solution for a more fine grained access to LokiStack logs on OpenShift Container Platform 4, ie, have the ability as the cluster administrator to grant access to logs on a namespace basis.
+
+The proposal provides an overview of the changes required to achieve this goal as well as some implementation details.
+
+## Motivation
+
+In enterprise environments, where OpenShift is used across different legal entities, it's common to have central teams that support the application teams in the respective entities. But given that some application may log sensitive data, those centralized support teams are not granted access to logs but they can only view specific objects, such as pods in the namespace.
+
+Currently, access to logs in LokiStack is granted when a user has access to the given namespace or when the user is part of a specific cluster-admin Group. And even though OpenShift Container Platform 4 does allow to configure RBAC to address this issue, LokiStack does not, and therefore grants access to logs to people that should not see them.
+
+This enhancement proposal presents a solution that enables cluster admins to have a more fine grained control over who accesses what logs.
+
+### User Stories
+
+* Use Case number one, is where a user has access to a specific namespace to see the objects included but is denied to view logs from pods (for legal reason mostly).
+
+* Use Case number two, is where a number of people have elevated permissions and therefore are able to access pretty much all namespaces on the OpenShift cluster. Usually those people are responsible for enabling and supporting applications when onboarding to OpenShift and therefore have permissions to see most of the objects in the given namespaces. However, due to legal and data protection constrains, those users can not have access to logs and therefore LokiStack should prevent them from seeing application specific logs.
+
+### Goals
+
+* Cluster admins can grant and revoke access of the workload logs of specific namespaces to users.
+* Cluster admins can grant and revoke access of the workload logs of specific namespaces to users with elevated privileges.
+
+### Non-Goals
+
+* Prevent users with elevated privileges from escalating their privileges to access application logs that they are not supposed to have access to. 
+
+## Proposal
```

Comment:
I question the need to modify the CL api to support this features.  Do we need to explicitly enable fine grained access control?  I would propose by moving towards what I have [proposed](https://github.com/openshift/enhancements/pull/1410/files#diff-9d07913b066ab6b25695e5fcd8e023e05ef0761832052d53cc76cf826f699448R81-R91) for multi clusterlogforwarding you may be able to "enable" the feature by creating the right permissions set.

How is it loki determines access?  Will it be by executing a SAR? If so, then I propose we can be more clever how we define roles by utilizing virtual resources.

Reply:
LokiStack is utilizing SARs against virtual resources already. Adding the switch is to keep backwards compatibility with the present implementation in CLO that gives all `system:authenticated` access to the logs by default. Of course it can be resolved without the switch proposed here by adding some upgrade code in CLO, but do users want to have full control of this update is not known upfront.

Reply:
> Of course it can be resolved without the switch proposed here by adding some upgrade code in CLO, but do users want to have full control of this update is not known upfront

This is what they are used to now and would be OOTB going forward.  Either way we are giving a knob to an admin which is either our spec or a set of RBAC commands.  I would contend it is better to give them instructions on "opting out" by using RBAC in lieu of adding a knob which changes the behavior of the operator by adding some binary decision